src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java

```diff
@@ -1,50 +1,117 @@
+    public static BlockHutGuardTower          blockHutGuardTower;
+    public static BlockHutWareHouse           blockHutWareHouse;
+    public static BlockConstructionTape       blockConstructionTape;
+    public static BlockConstructionTapeCorner blockConstructionTapeCorner;
+    public static BlockMinecoloniesRack       blockRack;
+    public static BlockTimberFrame            blockTimberFrame;
+    public static BlockWaypoint               blockWayPoint;
+    public static BlockHutBarracks            blockHutBarracks;
+    public static BlockHutBarracksTower       blockHutBarracksTower;
+    public static BlockInfoPoster             blockInfoPoster;
+    public static BlockPaperwall              blockPaperWall;
 
     // Deactivated for now
-    // public static final Block blockBarrel        = new BlockBarrel();
+    // public static  Block blockBarrel        = new BlockBarrel();
 
     /**
-     * private constructor to hide the implicit public one.
+     * Private constructor to hide the implicit public one.
      */
     private ModBlocks()
     {
+        /*
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Initates all the blocks. At the correct time.
+     */
+    public static void init(final IForgeRegistry<Block> registry)
+    {
+        blockConstructionTape = new BlockConstructionTape().registerBlock(registry);
+        blockConstructionTapeCorner = new BlockConstructionTapeCorner().registerBlock(registry);
+        blockHutBaker = (BlockHutBaker) new BlockHutBaker().registerBlock(registry);
+        blockHutBlacksmith = (BlockHutBlacksmith) new BlockHutBlacksmith().registerBlock(registry);
+        blockHutBuilder = (BlockHutBuilder) new BlockHutBuilder().registerBlock(registry);
+        blockHutCitizen = (BlockHutCitizen) new BlockHutCitizen().registerBlock(registry);
+        blockHutDeliveryman = (BlockHutDeliveryman) new BlockHutDeliveryman().registerBlock(registry);
+        blockHutFarmer = (BlockHutFarmer) new BlockHutFarmer().registerBlock(registry);
+        blockHutField = new BlockHutField().registerBlock(registry);
+        blockHutFisherman = (BlockHutFisherman) new BlockHutFisherman().registerBlock(registry);
+        blockHutGuardTower = (BlockHutGuardTower) new BlockHutGuardTower().registerBlock(registry);
+        blockHutLumberjack = (BlockHutLumberjack) new BlockHutLumberjack().registerBlock(registry);
+        blockHutMiner = (BlockHutMiner) new BlockHutMiner().registerBlock(registry);
+        blockHutStonemason = (BlockHutStonemason) new BlockHutStonemason().registerBlock(registry);
+        blockHutTownHall = (BlockHutTownHall) new BlockHutTownHall().registerBlock(registry);
+        blockHutWareHouse = (BlockHutWareHouse) new BlockHutWareHouse().registerBlock(registry);
+        blockSolidSubstitution = new BlockSolidSubstitution().registerBlock(registry);
+        blockSubstitution = new BlockSubstitution().registerBlock(registry);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'blockRack' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2444)